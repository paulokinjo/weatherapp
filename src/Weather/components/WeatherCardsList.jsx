import { useDispatch, useSelector } from 'react-redux';

import ArrowControl from '../../common/components/ArrowControl';
import React from 'react';
import Slider from 'react-slick';
import WeatherCard from './WeatherCard';
import { setCurrentPage } from '../../store/common/components/actions/arrowControlActions';

const WeatherCardsList = ({ weatherData }) => {
  const dispatch = useDispatch();
  const arrowControlState = useSelector((state) => state.arrow);

  const settings = {
    infinite: false,
    speed: 300,
    slidesToShow: 3,
    slidesToScroll: 1,
    swipeToSlide: true,
    focusOnSelect: true,
    nextArrow: (
      <ArrowControl
        customStyle={{
          top: '-80px',
          right: '25%',
          transform: 'rotate(360deg)',
        }}
        isVisible={arrowControlState.nextArrow}
        dispatch={dispatch}
      />
    ),
    prevArrow: (
      <ArrowControl
        customStyle={{ top: '-83px', left: '18%', transform: 'rotate(180deg)' }}
        isVisible={arrowControlState.prevArrow}
        dispatch={dispatch}
      />
    ),
    afterChange: (current) => dispatch(setCurrentPage(current)),
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 600,
        settings: {
          slidesToShow: 2,
          slidesToScroll: 1,
        },
      },
      {
        breakpoint: 480,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
        },
      },
    ],
  };

  return (
    <>
      <Slider {...settings}>
        {weatherData.map((data) => (
          <div key={data}>
            <WeatherCard info={data} />
          </div>
        ))}
      </Slider>
    </>
  );
};

export default WeatherCardsList;
