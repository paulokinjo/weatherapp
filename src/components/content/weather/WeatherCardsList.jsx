import ForwardIcon from '@material-ui/icons/Forward';
import React from 'react';
import Slider from 'react-slick';
import WeatherCard from './WeatherCard';

const list = [1, 2, 3, 4, 5];

function SampleNextArrow(props) {
  const { className, style, onClick } = props;
  return (
    <ForwardIcon
      className={className}
      style={{
        ...style,
        fontSize: 70,
        color: 'blue',
        top: -25,
        right: 20,
      }}
      onClick={onClick}
    />
  );
}

function SamplePrevArrow(props) {
  const { className, style, onClick } = props;
  return (
    <ForwardIcon
      className={className}
      style={{
        ...style,
        fontSize: 70,
        color: 'blue',
        transform: `rotate(180deg)`,
        top: -60,
        left: 30,
      }}
      onClick={onClick}
    />
  );
}

const settings = {
  infinite: false,
  speed: 300,
  slidesToShow: 3,
  slidesToScroll: 1,
  swipeToSlide: true,
  focusOnSelect: true,
  nextArrow: <SampleNextArrow />,
  prevArrow: <SamplePrevArrow />,
  responsive: [
    {
      breakpoint: 1024,
      settings: {
        slidesToShow: 3,
        slidesToScroll: 3,
      },
    },
    {
      breakpoint: 600,
      settings: {
        slidesToShow: 2,
        slidesToScroll: 2,
        initialSlide: 2,
      },
    },
    {
      breakpoint: 480,
      settings: {
        slidesToShow: 1,
        slidesToScroll: 1,
      },
    },
  ],
};

const WeatherCardsList = () => {
  return (
    <>
      <Slider {...settings}>
        {list.map((l) => (
          <div key={l}>
            <WeatherCard />
          </div>
        ))}
      </Slider>
    </>
  );
};

export default WeatherCardsList;
