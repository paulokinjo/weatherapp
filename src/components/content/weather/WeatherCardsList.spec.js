import React from 'react';
import WeatherCardsList from './WeatherCardsList';
import { shallow } from 'enzyme';

describe('WeatherCardsList', () => {
  const weatherCardsList = shallow(<WeatherCardsList />);

  describe('Layout', () => {
    it('should render properly', () => {
      expect(weatherCardsList).toMatchSnapshot();
    });
  });
});
