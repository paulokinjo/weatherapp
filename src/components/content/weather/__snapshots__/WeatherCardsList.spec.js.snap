// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardsList Layout should render properly 1`] = `ShallowWrapper {}`;
